File: study/그리디/05-모험가 길드.js
const fs = require('fs');
const input = fs.readFileSync('/dev/stdin').toString().split('\n');

const n = parseInt(input[0]);
const adventurers = input[1].split(' ').map((el) => parseInt(el));
let count = 0;

adventurers.sort((a, b) => b - a);

for (let i = 0; i < adventurers.length; i++) {
  i += adventurers[i];
  count++;
}

console.log(count);

File: study/그리디/05-모험가 길드.py
import sys
getInput = sys.stdin.readline

n = int(getInput())
adventurers = list(map(int, getInput().rstrip().split()))

adventurers.sort(reverse=True)
i = 0
count = 0

while True:
  i += adventurers[i]
  count += 1

  if i > len(adventurers)-1:
    break

print(count)